mender-auth/ipc/client.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-auth/ipc/client.hpp>
+
+#include <functional>
+#include <iostream>
+#include <string>
+
+#include <common/common.hpp>
+#include <common/error.hpp>
+#include <common/events_io.hpp>
+#include <common/http.hpp>
+#include <common/log.hpp>
+#include <common/path.hpp>
+
+namespace mender {
+namespace ipc {
+
+using namespace std;
+
+namespace path = mender::common::path;
+
+
+const IPCErrorCategoryClass IPCErrorCategory;
+
+const char *IPCErrorCategoryClass::name() const noexcept {
+	return "IPCErrorCategory";
+}
+
+string IPCErrorCategoryClass::message(int code) const {
+	switch (code) {
+	case NoError:
+		return "Success";
+	case SetupError:
+		return "Error during setup";
+	case RequestError:
+		return "HTTP client request error";
+	case ResponseError:
+		return "HTTP client response error";
+	case APIError:
+		return "API error";
+	case UnauthorizedError:
+		return "Unauthorized error";
+	default:
+		return "Unknown";
+	}
+}
+
+error::Error MakeError(IPCErrorCode code, const string &msg) {
+	return error::Error(error_condition(code, IPCErrorCategory), msg);
+}
+
+namespace type {
+namespace webserver {
+
+auto NoOpHeaderHandler = [](http::ExpectedIncomingRequestPtr exp_req) {};
+
+// Run a webserver and handle the events as requests
+error::Error Caching::ListenForEvents(
+	events::EventLoop &loop,
+	http::Server &server,
+	const string &server_url,
+	http::Client &client,
+	const conf::MenderConfig &config,
+	const string &private_key_path,
+	const string &identity_script_path) {
+	auto err = server.AsyncServeUrl(
+		server_url,
+		NoOpHeaderHandler,
+		[this, &loop, &client, &config, private_key_path, identity_script_path](
+			http::ExpectedIncomingRequestPtr exp_req) {
+			if (!exp_req) {
+				mlog::Error("Expected request failure: " + exp_req.error().message);
+				return;
+			}
+			auto req = exp_req.value();
+
+			auto exp_resp = req->MakeResponse();
+			if (!exp_resp) {
+				mlog::Error("Failed to create the response: " + exp_resp.error().message);
+				return;
+			}
+			auto resp = exp_resp.value();
+
+			mlog::Debug("Got path: " + req->GetPath());
+
+			if (req->GetPath() == "/getjwttoken") {
+				resp->SetHeader("X-MEN-JWTTOKEN", this->GetJWTToken());
+				resp->SetHeader("X-MEN-SERVERURL", this->GetServerURL());
```

Comment:
So there's no response data/payload only headers?

Reply:
Yes. I figured it was the easiest approach, but you being the consumer, feel free to tell me otherwise

Reply:
I thought this was part of some public/documented API.

Reply:
No, the public/documented API is DBus